### PR TITLE
Add Speed Limit Requests

### PR DESCRIPTION
**Pull request by PranayShirodkar**
### continuous_checker

- Add continuous_checker param to LaneBlocker node. This param is a bool with default value false.
- If obstacles are static, continuous_checker should be left as false. If obstacles are dynamic/moving, set continuous_checker to true.
- When continuous_checker is false, the LaneBlocker node will only check an obstacle for intersection with all lanes when the obstacle first appears.
- When continuous_checker is true, the LaneBlocker node will continuously check an obstacle for intersection with all lanes as long as the obstacle is present.

### add speed limits

- Lanes can now be in Closed state, Speed Limited state and normal state
- Refactored the lane open and lane close msg creation and sending
- Add speed limit and speed unlimit msg
- Handle all 6 possible state transitions
- Add python bindings for agv::FleetUpdateHandle::limit_lane_speeds() and agv::FleetUpdateHandle::remove_speed_limits()  - related PR in [rmf_demos](https://github.com/open-rmf/rmf_demos/pull/152)

### CODE REVIEW

**Comment by mxgrey** [comment]
@Yadunund since this is targeting your draft PR #210 I'll leave it up to you if/when/how this gets merged.

**Review by Yadunund — APPROVED** [review]
I've left TODOs for things to address before merging into `main`. Fixed const correctness and made an API tweak to the publisher functions to prevent segfaults if the unordered_maps containing the unique_ptrs are accessed after moving the messaging during `publish()`.

**Pull request merged**
